### PR TITLE
Pass Mappings To Update Helper

### PR DESCRIPTION
**Pull request by Mikep86**
This PR addresses a dependency needed in https://github.com/elastic/elasticsearch/pull/112319. In that PR, we need to access to a `MappingLookup` instance in `UpdateHelper#extractGetResult` so that we can filter out vector fields from source when handling update requests.

I separated this out into a dedicated PR to reduce the scope of https://github.com/elastic/elasticsearch/pull/112319 and make this change easier to review.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java [hunk 1]
```diff
@@ -479,7 +480,14 @@ && isConflictException(executionResult.getFailure().getCause())
         }
         final BulkItemResponse response;
         if (isUpdate) {
-            response = processUpdateResponse((UpdateRequest) docWriteRequest, context.getConcreteIndex(), executionResult, updateResult);
+            final MappingLookup mappingLookup = context.getPrimary().mapperService().mappingLookup();
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java:483]: As this is the `TransportShardBulkAction`, I don't think there's the possibility of having a null `MapperService` or `MappingLookup`. Do you think we should be defensive about that, or can we be sure that they're not null?

**Review by carlosdelest — APPROVED** [review]
LGTM - one question about ensuring mapper service / lookup can be null or not 👍

File: server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java (hunk 1 above)
Comment (Mikep86) [reply server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java:483]: Agreed, `MapperService` and `MappingLookup` should always be non-null here. Given we're getting this from an `IndexShard` instance, I don't think we need to be _that_ defensive; how about an `assert` so we catch it in tests if it's a problem?

Comment (carlosdelest) [reply server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java:483]: > how about an assert so we catch it in tests if it's a problem?

I like that, thanks Mike!

**Review by kderusso — APPROVED** [review]
LGTM!

**Pull request merged**